sherpa/astro/data.py

```diff
@@ -935,9 +935,13 @@ def __init__(self, name, energ_lo, energ_hi, specresp, bin_lo=None,
         self._lo, self._hi = energ_lo, energ_hi
         self.energ_lo = energ_lo
         self.energ_hi = energ_hi
-        Data1DInt.__init__(self, name, energ_lo, energ_hi, specresp)
 
-    def _repr_html_(self):
+        super().__init__(name, energ_lo, energ_hi, specresp)
+
+        self._xlabel = 'Energy (keV)'
+        self._ylabel = None
```

Comment:
May I ask why you chose to set this in the `__init__` call instead of adding them as class attributes? The latter works better for inheritance IMO and is easier to read and set - you could derive a new class that changes the labels without the need to write a new `__init__` method.
Of course, this works, too, so just asking.

Reply:
That's true specifically when init does not call super. Or is that the reason you did not do it? Because our class hierachy is somewhat messed up? Or is it that class attributes can't be overridden later by instance attributes because of the `NoNewAttributesAfterInit` as a base class?

Reply:
a) I never know what the intended semantic difference is between class and object attributes for Python.

b) our inheritance here is perhaps not what you might think it is, which makes reasoning about some of this interesting - I'd have to look into it to see how much it saves here - e.g. it would hopefully avoid the need for  

https://github.com/sherpa/sherpa/pull/2086/files#diff-83c9f3e1eca305239432be420691a7cb613f4321aa77549f0e33c8ec041c23daR2096-R2107